cmd/groups-auto-updater/main.go

```diff
@@ -0,0 +1,176 @@
+			if err := config.FromPeribolosConfig.validate(); err != nil {
+				return fmt.Errorf("from_peribolos_config[%s]: %v", group, err)
+			}
+		}
+	}
+	return nil
+}
+
+// GroupConfig holds the information of the group configuration
+type GroupConfig struct {
+	FromPeribolosConfig *FromPeribolosConfig `json:"from_peribolos_config,omitempty"`
+}
+
+// FromPeribolosConfig has the peribolos config file and from which org to get the users from.
+type FromPeribolosConfig struct {
+	ConfigFile string `json:"config_file"`
+	Org        string `json:"org"`
+}
+
+func (f *FromPeribolosConfig) validate() error {
+	if len(f.ConfigFile) == 0 {
+		return errors.New("config_file is not specified")
+	}
+
+	if len(f.Org) == 0 {
+		return errors.New("org is not specified")
+	}
+	return nil
+}
+
+type options struct {
+	configFile string
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fs.StringVar(&o.configFile, "config", "", "Configuration file path")
+
+	fs.Parse(os.Args[1:])
+	return o
+}
+
+func validateOptions(o options) error {
+	if len(o.configFile) == 0 {
+		return fmt.Errorf("--config was not provided")
+	}
+	return nil
+}
+
+func loadConfig(configFile string) (*Config, error) {
+	var config *Config
+	b, err := ioutil.ReadFile(configFile)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't read configuration file: %v", err)
+	}
+
+	if err := yaml.Unmarshal(b, &config); err != nil {
+		return nil, fmt.Errorf("couldn't unmarshal configuration file: %v", err)
+	}
+
+	if err := config.validate(); err != nil {
+		return nil, fmt.Errorf("couldn't validate configuration file: %v", err)
+	}
+
+	return config, nil
+}
+
+func getUserV1Client() (*userV1.UserV1Client, error) {
+	clusterConfig, err := util.LoadClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("could not load cluster clusterConfig: %v", err)
+	}
+
+	userV1Client, err := userV1.NewForConfig(clusterConfig)
+	if err != nil {
+		return nil, fmt.Errorf("could not create user openshift client: %v", err)
+	}
+
+	return userV1Client, nil
+}
+
+func getUsersFromPeribolosConfig(pConfig *FromPeribolosConfig) (sets.String, error) {
+	users := sets.NewString()
+
+	b, err := ioutil.ReadFile(pConfig.ConfigFile)
+	if err != nil {
+		return users, fmt.Errorf("could not read peribolos configuration file: %v", err)
+	}
+
+	var peribolosConfig org.FullConfig
+	if err := yaml.Unmarshal(b, &peribolosConfig); err != nil {
+		return users, fmt.Errorf("failed to unmarshal peribolos config: %v", err)
+	}
+
+	users.Insert(peribolosConfig.Orgs[pConfig.Org].Admins...)
+	users.Insert(peribolosConfig.Orgs[pConfig.Org].Members...)
+
+	return users, nil
+}
+
+func updateGroup(groupName string, groupConfig GroupConfig, groupV1Interface userV1.GroupInterface) error {
+	g, err := groupV1Interface.Get(groupName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("couldn't get group from cluster: %v", err)
+	}
+
+	if groupConfig.FromPeribolosConfig != nil {
```

Comment:
Or, this ... why does our tool have to handle the case where we are explicitly configured to do nothing?